crates/router/src/db/merchant_connector_account.rs

```diff
@@ -1,13 +1,99 @@
-use error_stack::IntoReport;
+use common_utils::ext_traits::{ByteSliceExt, Encode};
+use error_stack::{IntoReport, ResultExt};
 use masking::ExposeInterface;
 
 use super::{MockDb, Store};
 use crate::{
     connection::pg_connection,
     core::errors::{self, CustomResult},
-    types::storage,
+    services::logger,
+    types::{self, storage},
 };
 
+#[async_trait::async_trait]
+pub trait ConnectorAccessToken {
+    async fn get_access_token(
+        &self,
+        merchant_id: &str,
+        connector_name: &str,
+    ) -> CustomResult<Option<types::AccessToken>, errors::StorageError>;
+
+    async fn set_access_token(
+        &self,
+        merchant_id: &str,
+        connector_name: &str,
+        access_token: types::AccessToken,
+    ) -> CustomResult<(), errors::StorageError>;
+}
+
+#[async_trait::async_trait]
+impl ConnectorAccessToken for Store {
+    async fn get_access_token(
+        &self,
+        merchant_id: &str,
+        connector_name: &str,
+    ) -> CustomResult<Option<types::AccessToken>, errors::StorageError> {
+        //TODO: Handle race condition
+        // This function should acquire a global lock on some resource, if access token is already
+        // being refreshed by other request then wait till it finishes and use the same access token
+        let key = format!("{merchant_id}_{connector_name}");
```

Comment:
we need to add feature prefix while fetching from config map

Reply:
This is reading directly from redis. Do we need that here?

Reply:
yes, for eg: mid123_stripe, without feature, any feature can override this value and we will runtime bug